src/render_context.rs

```diff
@@ -58,6 +59,20 @@ impl<'a> RenderContext<'a> {
             }
         }
     }
+    /// return an empty rendering context
+    pub_unstable_internal! {
+        pub(crate) fn empty() -> Self {
+        let cached_set = &RefCell::new(CachedSet::default());
```

Comment:
Nitpick: please indent code properly. Everything within a function body should be 4 spaces indented.